academic-services/knowledge-exploration-service/reference-index-data.md

```diff
@@ -0,0 +1,387 @@
+string | string | `"Ti": "an overview of microsoft academic service mas and applications"`
+
+### Multiple attribute values
+
+To define more than one value for an attribute you can either specify the attribute multiple times, e.g.:
+
+``` JSON
+{ "RId": 123, "RId": 456 }
+{ "F": { "FN": "machine learning", "FId": 119857082 }, "F": { "FN": "artificial intelligence", "FId": 154945302 } }
+```
+
+Or use JSON arrays, e.g.:
+
+``` JSON
+{ "RId": [ 123, 456 ] }
+{ "F": [ { "FN": "machine learning", "FId": 119857082 }, { "FN": "artificial intelligence", "FId": 154945302 } ] }
+```
+
+## Entity log probability
+
+An important component of the KES top-N retrieval algorithm is the concept of entity probability, which informs KES of each entities relative importance. Entity probability can be understood as the likelihood of it being named the most important in a survey conducted on all members its network/graph ([see "Saliency: An Eigencentrality Measure for Heterogeneous Dynamic Network" in "A Review of Microsoft Academic Services for Science of Science Studies"](https://www.frontiersin.org/articles/10.3389/fdata.2019.00045/pdf)).
+
+KES allows this to be defined for an entity as a relative log probability using the optional "logprob" attribute, which is an additional attribute that is always available and not defined in the [schema](reference-index-schema.md). Given a probability *p* between 0 and 1, the corresponding log probability can be computed as log(*p*), where log() is the natural log function. When no value is specified for logprob, the default value 0 is used.
+
+## Entity data file format
+
+An entity data file contains one or more rows/lines of [entity data](#entity-data-json-format), and is used as the "data file" when [building an index](reference-index-build.md).
+
+> [!IMPORTANT]
+> While each **row** in an entity data file must be a valid JSON object, the overall file itself is not JSON but simply a *rows/lines of JSON objects*. The file contents are **NOT** meant to be an array of JSON objects.
+
+> [!TIP]
+> If your entities use a log probability score, you can significantly increase the performance of indexing by pre-sorting the entity data by the log probability score in descending order (entity with highest log probability in first row, entity with lowest log probability in last row).
+
+## Example
+
+### Academic paper entity
+
+The following entity JSON data is for an index that supports academic paper entities, and is part of the larger [MAKES customization tutorial](tutorial.md) and uses a subset ([defined in the schema reference](reference-index-schema.md#example)) of the full [MAKES academic entity schema](reference-makes-api-entity-schema.md).
+
+> [!IMPORTANT]
+> The following example is purposefully using tabular formatting of the JSON data to make it easy to understand. However in practice **this data would have no such formatting and be contained on a *single* line**.
+
+``` JSON
+{
+  "logprob": -19.425,
```

Comment:
weight

Reply:
See my explanation above

---
In reply to: [455280592](https://github.com/MicrosoftDocs/microsoft-academic-services/pull/164#discussion_r455280592) [](ancestors = 455280592)